src/dynamics/spectral_grid.jl

```diff
@@ -117,14 +123,91 @@ function Base.show(io::IO, SG::SpectralGrid)
       print(io, "└ Device:     $(typeof(device)) using $ArrayType")
 end
 
+# Constructor that takes all [OPTION] parameters as keyword arguments
+# and calculates all derived fields
+function SpectralGrid(;
+    NF::Type{<:AbstractFloat} = DEFAULT_NF,
+    device::AbstractDevice = DEFAULT_DEVICE,
+    ArrayType::Type{<:AbstractArray} = default_array_type(device),
+    trunc::Int = DEFAULT_TRUNC,
+    Grid::Type{<:AbstractGrid} = DEFAULT_GRID,
+    dealiasing::Number = 2.0,
+    radius::Number = DEFAULT_RADIUS,
+    nparticles::Int = 0,
+    nlayers::Int = DEFAULT_NLAYERS,
+    nlayers_soil::Int = DEFAULT_NLAYERS_SOIL
+)
+    # Convert numeric parameters to Float64
+    dealiasing_f64 = Float64(dealiasing)
+    radius_f64 = Float64(radius)
+    
+    # Calculate derived fields
+    VectorType = ArrayType{NF, 1}
+    MatrixType = ArrayType{NF, 2}
+    TensorType = ArrayType{NF, 3}
+    
+    # Spectral space
+    spectrum = Spectrum(trunc+2, trunc+1)
```

Comment:
take on `architecture`?

Reply:
Will be done in #727 , we don't have the architectures here yet.